packages/forms/src/hooks/UseTextInput.ts

```diff
@@ -0,0 +1,66 @@
+import { FocusEventHandler, RefObject, useCallback } from "react";
+import { TextInputProps } from "../components/ui/text-input/TextInput";
+import { useKeyboardNavigation } from "./UseKeyboardNavigation";
+import { useSelectAllOnFocus } from "./UseSelectAllOnFocus";
+import { useSelectAllOnMount } from "./UseSelectAllOnMount";
+
+export const useTextInput = (
+  ref: RefObject<HTMLInputElement>,
+  {
+    onEnter,
+    onEsc,
+    onChange,
+    onValueChange,
+    selectAllOnFocus,
+    selectAllOnMount,
+    moveCursorToEndOnMount,
+    onDone,
+    onMove,
+    onFocus,
+    onBlur,
+    onKeyDown
+  }: TextInputProps
+) => {
+  const { onFocusHandler } = useSelectAllOnFocus(
+    ref,
+    onFocus,
+    selectAllOnFocus
+  );
+  useSelectAllOnMount(ref, !!moveCursorToEndOnMount, !!selectAllOnMount);
+
+  const { onKeyDownHandler, wasCancelled } = useKeyboardNavigation(
+    ref,
+    onKeyDown,
+    onEnter,
+    onEsc,
+    onMove
+  );
+
+  const onBlurHandler: FocusEventHandler<HTMLInputElement> = ev => {
+    if (onDone && !wasCancelled) {
+      onDone(ev.target.value || "");
```

Comment:
Do we want to use ?? instead of ||?

Reply:
Yes.
This is also existing code that has been moved into hooks.
But I've updated it.